frontend/src/data/variables/AcsPopulationProvider.ts

```diff
@@ -77,24 +74,22 @@ class AcsPopulationProvider extends VariableProvider {
         .resetIndex();
     }
 
-    const acsStandard = acsNonStandard.where((row) =>
-      standardizedRaces.includes(row.race_and_ethnicity)
-    );
-    if (breakdowns.demographic === "race" && breakdowns.geography === "state") {
-      return acsStandard;
-    }
-
-    if (
-      breakdowns.demographic === "race" &&
-      breakdowns.geography === "national"
-    ) {
-      return acsStandard
-        .pivot("race_and_ethnicity", {
-          state_fips: (series) => USA_FIPS,
-          state_name: (series) => USA_DISPLAY_NAME,
-          population: (series) => series.sum(),
-        })
-        .resetIndex();
+    switch (breakdowns.demographic) {
+      case "race_nonstandard":
+        return breakdowns.geography === "national"
+          ? createNationalTotal(acsDataFrame, "race_and_ethnicity")
+          : acsDataFrame;
+      case "race":
+        const standardizedAcsData = acsDataFrame.where((row) =>
+          standardizedRaces.includes(row.race_and_ethnicity)
+        );
+        return breakdowns.geography === "national"
+          ? createNationalTotal(standardizedAcsData, "race_and_ethnicity")
+          : standardizedAcsData;
+      case "age":
+        return breakdowns.geography === "national"
+          ? createNationalTotal(acsDataFrame, "age")
+          : acsDataFrame;
```

Comment:
This is very clean, I like the structure!

Reply:
:)

Reply:
:)